File: baekjoon_all/01000+/boj_1978_solution1.py
# Solve 2022-05-24
# Update 2023-03-01

import sys

input = lambda : sys.stdin.readline().rstrip()

MAX_SIZE = 1001
primes = [True] * MAX_SIZE
primes[0] = primes[1] = False
for i in range(2, MAX_SIZE):
    if primes[i]:
        for j in range(i + i, MAX_SIZE, i):
            primes[j] = False

n = int(input())
li = list(map(int, input().split()))

ans = 0
for i in li:
    if primes[i]:
        ans += 1

print(ans)

File: baekjoon_all/01000+/boj_1978_solution1.cpp
// Solve 2023-03-01

#include <bits/stdc++.h>
using namespace std;

#ifdef BOJ
#define BOJTEST(x) ((void)0)
#else
#define BOJTEST(x) cout << "[Debug] " << #x << ':' << x << '\n'
#endif
#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;
using uint = unsigned int;
using ull = unsigned long long;

const int MAX_SIZE = 1001;

int main() {
    FASTIO;

    vector<bool> primes(MAX_SIZE, true);
    primes[0] = primes[1] = false;
    for (int i = 2; i < MAX_SIZE; i++) {
        if (primes[i]) {
            for (int j = i + i; j < MAX_SIZE; j += i) {
                primes[j] = false;
            }
        }
    }

    int n;
    cin >> n;

    int ans = 0;
    for (int i = 0; i < n; i++) {
        int x;
        cin >> x;
        if (primes[x]) {
            ans++;
        }
    }
    cout << ans << '\n';

    return 0;
}